cosi.proto

```diff
@@ -97,22 +150,13 @@ message ProvisionerCreateBucketRequest {
 
     // This field is OPTIONAL
     // Protocol specific information required by the call is passed in as key,value pairs.
-    map<string,string> bucket_context = 2;
-
-    enum AnonymousBucketAccessMode {
-        // Default, disallow uncredentialed access to the backend storage.
-	BUCKET_PRIVATE = 0;
-        // Read only, uncredentialed users can call ListBucket and GetObject.
-	BUCKET_READ_ONLY = 1;
-        // Write only, uncredentialed users can only call PutObject.
-	BUCKET_WRITE_ONLY = 2;
-        // Read/Write, same as ro with the addition of PutObject being allowed.
-	BUCKET_READ_WRITE = 3;
-    }
+    map<string,string> parameters = 2;
     
     // This field is OPTIONAL
     // Allow uncredentialed access to bucket.
     AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+
+    Protocol protocol = 4;
 }
 
 message ProvisionerCreateBucketResponse {
```

Comment:
Shouldn't this return a `Bucket` struct which at a very minimum should contain `string bucket_id`?

Reply:
bucket_id is already the unique id

Reply:
You should reconsider this. Don't tie together the suggested name from the actual final identifier. `bucket_name` on create bucket request serves as a idempotency token and as suggestion for the name (which the initial set of vendors may all just use as is without modifying). And on create bucket response, vendor must return the actual unique `bucket_id` which may be the same or different from `bucket_name`.